cmd/thermal-recorder/snapshot.go

```diff
@@ -57,3 +60,42 @@ func newSnapshot(lastFrame int) (*cptvframe.Frame, error) {
 	}
 	return f, nil
 }
+
+func newSnapshotRecording() error {
+	mu.Lock()
+	defer mu.Unlock()
+
+	if processor == nil {
+		return errors.New("reading from camera has not started yet")
+	}
+
+	processor.StartSnapshot = true
+	return nil
+}
+
+func snapshotRecordingTriggers(processor *motion.MotionProcessor, window window.Window) {
```

Comment:
Would it not be easier to put the logic for this in processSnapshot function? rather than having a process/thread sitting waiting for ~12 hours?

Reply:
Putting it in `processSnapshot` it would be calculated each frame where here it is just calculated once then is just waiting in the goroutine .